synapse/api/auth.py

```diff
@@ -149,6 +149,42 @@ async def get_user_by_req(
                 is invalid.
             AuthError if access is denied for the user in the access token
         """
+        parent_span = active_span()
+        with start_active_span("get_user_by_req"):
+            requester = await self._wrapped_get_user_by_req(
+                request, allow_guest, rights, allow_expired
+            )
+
+            if parent_span:
+                if requester.authenticated_entity in self._force_tracing_for_users:
+                    # request tracing is enabled for this user, so we need to force it
+                    # tracing on for the parent span (which will be the servlet span).
+                    #
+                    # It's too late for the get_user_by_req span to inherit the setting,
+                    # so we also force it on for that.
+                    force_tracing()
+                    force_tracing(parent_span)
+                parent_span.set_tag(
+                    "authenticated_entity", requester.authenticated_entity
+                )
```

Comment:
I had to convince myself that `authenticated_entity` was the same here:

* For appservices `authenticated_entity` isn't set during creation of the requester and we report the tag as `user_id`; this was essentially inlined logic from [`create_requester`](https://github.com/matrix-org/synapse/blob/ecfcd9bbbed62f8ca5d42bd8ba02b696e9079573/synapse/types.py#L183-L184)
* For non-appservices, this was set to `user_info.token_owner`, which is the same thing that is passed as the `authenticated_entity` in `create_requester`.

Reply:
sorry, yes, I must have gone through that same thought process, but forgot to write it down here!

Reply:
No problem! Definitely better to put all this logic in one place though!